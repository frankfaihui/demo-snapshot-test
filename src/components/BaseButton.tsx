import styled from "@emotion/styled";
import { ReactNode } from "react";

// Button props
export interface BaseButtonProps {
  className?: string;
  children?: ReactNode;
  onClick?: () => void;
}

// Base Button component
const StyledButton = styled.button<BaseButtonProps>({
  display: "block",
  border: "1px solid #e2e8f0",
  color: "black",
  padding: "10px",
  backgroundColor: "white",
});

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const NewButton = styled(StyledButton)({
  backgroundColor: "green",
});

export default function BaseButton({ className, children, ...props }: BaseButtonProps) {

  // uncomment below and run 'npm run test` to see the error, then run `npm run test` --update to see the updated snapshot
  // return (
  //   <NewButton className={className} {...props}>
  //     <div>
  //       {children}
  //     </div>
  //   </NewButton>
  // )
  return <StyledButton className={className} {...props}>{children}</StyledButton>;
} 