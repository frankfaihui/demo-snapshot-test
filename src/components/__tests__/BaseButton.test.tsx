import React from 'react';
import { render } from '@testing-library/react';
import BaseButton from '../BaseButton';

describe('BaseButton', () => {
  it('renders basic button correctly', () => {
    const { container } = render(<BaseButton>Click me</BaseButton>);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 