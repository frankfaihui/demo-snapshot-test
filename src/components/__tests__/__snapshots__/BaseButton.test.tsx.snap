// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton renders basic button correctly 1`] = `
.emotion-0 {
  display: block;
  border: 1px solid #e2e8f0;
  color: black;
  padding: 10px;
  background-color: white;
}

<button
  class="emotion-0"
>
  Click me
</button>
`;
